<article>
  <section>
    <!-- <h2>Citations</h2> -->

    <!-- Any published work which has made use of RNApdbee should cite the following paper: -->
    <ul>
      <!-- <li>RNApdbee 2.0: Multifunctional Tool for RNA Structure Annotation. T. Zok, M. Antczak, M. Zurkowski, M. Popenda,
        J. Blazewicz, R.W. Adamiak, M. Szachniuk. <i>Nucleic Acids Research</i>. 2018. 46(W1):W30–W35. doi:<a
          href="https://doi.org/10.1093/nar/gky314">10.1093/nar/gky314</a></li> -->
          <li> T. Zok, M. Antczak, M. Zurkowski, J. Pielesiak, M. Popenda,J. Blazewicz, R.W. Adamiak, 
            M. Szachniuk. RNApdbee 3.0: Working Paper. <i>Journal</i>. 2026. 46(W1):W30–W35. doi:<a
          href="https://doi.org">doi link when published</a></li>
    </ul>

    <!-- In addition, please cite the papers of the integrated tools' authors: -->
    <!-- Papers related to third-party algorithms integrated in RNApdbee 3.0:
    <ul>
      <li>Contact Networks in RNA: P. Roy, D. Bhattacharyya. A Structural Bioinformatics Study with a New Tool.
        <i>Journal of Computer-Aided Molecular Design</i>. 2022. 36(2):131–140. doi:<a
          href="https://doi.org/10.1007/s10822-021-00438-x">10.1007/s10822-021-00438-x</a></li>
      <li>R-Chie:  V. Tsybulskyi, M. Mounir, I.M. Meyer. A Web Server and R Package for Visualizing 
        Cis and Trans RNA–RNA, RNA–DNA and DNA–DNA Interactions. <i>Nucleic Acids Research</i>. 2020. 48(18):e105–e105. doi:<a
          href="https://doi.org/10.1093/nar/gkaa708">10.1093/nar/gkaa708</a></li>
      <li>RNApuzzler:  D. Wiegreffe, D. Alexander, P.F. Stadler, D. Zeckzer. 
        Efficient Outerplanar Drawing of RNA-Secondary Structures.<i>Bioinformatics</i>. 2019. 35(8):1342–1349. doi:<a
          href="https://doi.org/10.1093/bioinformatics/bty817">10.1093/bioinformatics/bty817</a></li>
      <li>Barnaba:  S. Bottaro, G. Bussi, G. Pinamonti, S. Reißer, W. Boomsma, K. Lindorff-Larsen. 
        Software for Analysis of Nucleic Acid Structures and Trajectories.<i>RNA</i>. 2019. 25(2):219–231. doi:<a
          href="https://doi.org/10.1261/rna.067678.118">10.1261/rna.067678.118</a></li>
      <li>ViennaRNA Package 2.0: R. Lorenz, S.H. Bernhart, C. Höner zu Siederdissen, H. Tafer, C. Flamm, P.F. Stadler,
        I.L. Hofacker. ViennaRNA Package 2.0. <i>Algorithms for Molecular Biology</i>. 2011. 6(1):26. doi:<a
          href="https://doi.org/10.1186/1748-7188-6-26">10.1186/1748-7188-6-26</a></li>
      <li>VARNA:  K. Darty, A. Denise, Y. Ponty. Interactive Drawing and Editing of the RNA Secondary Structure.
        <i>Bioinformatics</i>. 2009. 25(15):1974–1975. doi:<a
          href="https://doi.org/10.1093/bioinformatics/btp250">10.1093/bioinformatics/btp250</a></li>
      <li>PseudoViewer3:  Y. Byun, K. Han.Generating Planar Drawings of Large-Scale RNA Structures with Pseudoknots.
        <i>Bioinformatics</i>. 2009. 25(11):1435–1437. doi:<a
          href="https://doi.org/10.1093/bioinformatics/btp252">10.1093/bioinformatics/btp252</a></li>
      <li>FR3D:  M. Sarver, C.L. Zirbel, J.Stombaugh, A. Mokdad, N.B. Leontis. Finding Local and Composite Recurrent 
        Structural Motifs in RNA 3D Structures. <i>Journal of Mathematical Biology</i>. 2008. 56(1–2):215–252. doi:<a
          href="https://doi.org/10.1007/s00285-007-0110-x">10.1007/s00285-007-0110-x</a></li>
      <li>WebLogo:  G.E. Crooks, G. Hon, J.-M. Chandonia, S.E. Brenner. A Sequence Logo Generator.<i>Genome
          Research</i>. 2004. 14(6):1188–1190. doi:<a href="https://doi.org/10.1101/gr.849004">10.1101/gr.849004</a>
      </li>
      <li> H. Yang. Tools for the Automatic Identification and Classification of RNA Base Pairs.<i>Nucleic Acids
          Research</i>. 2003. 31(13):3450–3460. doi:<a href="https://doi.org/10.1093/nar/gkg529">10.1093/nar/gkg529</a>
      </li>
      <li> P. Gendron, S. Lemieux, F. Major.Quantitative Analysis of Nucleic Acid Three-Dimensional Structures.
        <i>Journal of Molecular Biology</i>. 2001. 308(5):919–936. doi:<a
          href="https://doi.org/10.1006/jmbi.2001.4626">10.1006/jmbi.2001.4626</a></li>
    </ul> -->

    Other articles related to RNApdbee:
    <ul>
      <li>BioCommons:  T. Zok. <i>Bioinformatics</i>. 2021.
       A Robust Java Library for RNA Structural Bioinformatics. 37(17):2766–2767. doi:<a
          href="https://doi.org/10.1093/bioinformatics/btab069">10.1093/bioinformatics/btab069</a></li>
      <li> T. Zok, J. Badura, S. Swat, K. Figurski, M. Popenda, M. Antczak. New Models and Algorithms for RNA Pseudoknot Order Assignment.
        <i>International Journal of Applied Mathematics and Computer Science</i>. 2020.
        30(2):315–324. doi:<a href="https://doi.org/10.34768/amcs-2020-0024">10.34768/amcs-2020-0024</a></li>
      <li> M. Antczak, M. Popenda, T. Zok, M. Zurkowski, R.W. Adamiak, M. Szachniuk. 
        New Algorithms to Represent Complex Pseudoknotted RNA Structures in Dot-Bracket Notation.<i>Bioinformatics</i>. 2018. 34(8):1304–1312. doi:<a
          href="https://doi.org/10.1093/bioinformatics/btx783">10.1093/bioinformatics/btx783</a></li>
      <li> M. Antczak, T. Zok, M. Popenda, P. Lukasiak, R.W. Adamiak, J. Blazewicz, M. Szachniuk. 
        RNApdbee – a Webserver to Derive Secondary Structures from Pdb Files of Knotted and Unknotted RNAs.<i>Nucleic Acids
          Research</i>. 2014. 42(W1):W368–W372. doi:<a href="https://doi.org/10.1093/nar/gku330">10.1093/nar/gku330</a>
      </li>
    </ul>
  </section>
</article>
